tests/Matcher/ExtractorFactoryTest.php

```diff
@@ -80,4 +80,72 @@ public function testGetPathWithBasePathThatDoesNotMatch()
         $extractor = $extractorFactory->createPathExtractor();
         $this->assertSame('', $extractor($this->request));
     }
+
+    public function testGetHeaderWithExistingHeader()
+    {
+        $request = new Request(
+            [],
+            [],
+            [],
+            [],
+            [],
+            ['HTTP_CONTENT_TYPE' => 'application/json']
+        );
```

Comment:
Note that I instantiated a real Symfony request here, that way we can avoid mocking them (as is done in the other test cases). Unless I am missing something, this should be preferred IMO, as it does not couple the tests to the implementation. :)